evap/staff/importers/base.py

```diff
@@ -56,11 +62,29 @@ class Category(Enum):
     message: str
 
 
+@dataclass
+class ImporterDecisionEntry(ImporterLogEntry):
+    existing: str
+    imported: str
+    user_object: UserProfile
+    user_data: "UserData"
+    resolved: bool = False
+
+    def apply(self):
```

Comment:
"Decision" and "apply" do not match.

When thinking about how we could structure this, my mind went through compiler errors / messages and corresponding autofix features. I wonder how clangd manages this internally.

I'd not be opposed of changing our model to that of a compiler. We could rename "Log" to "Diagnostics", where each entry is associated with a checker, and checkers have a "autofix" feature. I guess the most annoying part is that checkers by design work on a heavily preprocessed representation to make them as simple as possible. We'd always have to re-build the `xlsx` file from scratch in the end instead of modifying an existing one. I wonder how clangd and similar do this internally.

Reply:
Another thought, even more annoying: Compiler autofixes sometimes fail / do weird stuff. In an IDE, that's usually not a problem, as I can observe the change right on my screen, and git / PR review will also notice a shady transformation.

With our current approach, users will not get a nice diff view, so they'll likely not notice if anything went wrong.